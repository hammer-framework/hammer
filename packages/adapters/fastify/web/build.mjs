import { build, defaultIgnorePatterns } from '../../../../buildDefaults.mjs'

await build({
  entryPointOptions: {
    ignore: [...defaultIgnorePatterns, './src/types.ts'],
  },
})
