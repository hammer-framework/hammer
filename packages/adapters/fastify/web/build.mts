import { build, defaultBuildOptions } from '@redwoodjs/framework-tools'

await build({
  buildOptions: {
    ...defaultBuildOptions,
    bundle: true,
    entryPoints: ['./src/web.ts'],
    packages: 'external',
  },
})
