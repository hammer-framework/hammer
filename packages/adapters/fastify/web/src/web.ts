import fs from 'fs'
import path from 'path'

import httpProxy from '@fastify/http-proxy'
import fastifyStatic from '@fastify/static'
import fastifyUrlData from '@fastify/url-data'
import fg from 'fast-glob'
import type { FastifyInstance, HookHandlerDoneFunction } from 'fastify'

import { getConfig, getPaths } from '@redwoodjs/project-config'

export interface RedwoodFastifyWebOptions {
  redwood?: {
    apiUrl?: string
    apiProxyTarget?: string

    /**
     * @deprecated Use `apiProxyTarget` instead.
     */
    apiHost?: string
  }
}

export async function redwoodFastifyWeb(
  fastify: FastifyInstance,
  opts: RedwoodFastifyWebOptions,
  done: HookHandlerDoneFunction
) {
  fastify.log.debug('opts')
  fastify.log.debug(opts)

  const options = resolveOptions(opts)

  fastify.log.debug('resolving options')
  fastify.log.debug(options)

  await fastify.register(fastifyUrlData)

  // Serve prerendered files directly, instead of the index
  const prerenderedFiles = await fg('**/*.html', {
    cwd: getPaths().web.dist,
    ignore: ['index.html', '200.html', '404.html'],
  })

  for (const prerenderedFile of prerenderedFiles) {
    const [pathName] = prerenderedFile.split('.html')

    fastify.get(`/${pathName}`, (_, reply) => {
      reply.header('Content-Type', 'text/html; charset=UTF-8')
      reply.sendFile(prerenderedFile)
    })
  }

  fastify.log.debug(`registering static assets at ${getPaths().web.dist}`)
  fastify.log.debug(getPaths().web.dist)
  // Serve static assets
  fastify.register(fastifyStatic, {
    root: getPaths().web.dist,
  })

  // If `apiProxyTarget` is set, proxy requests from `apiUrl` to `apiProxyTarget`.
  // In this case, `apiUrl` has to be relative; `resolveOptions` above throws if it's not
  if (options.redwood.apiProxyTarget) {
    fastify.log.debug('registering proxy')

    fastify.register(httpProxy, {
      prefix: options.redwood.apiUrl,
      upstream: options.redwood.apiProxyTarget,
      disableCache: true,
    })
  }

  // If `200.html` exists, the project has been prerendered.
  // If it doesn't, fallback to the default (`index.html`)
  const prerenderIndexPath = path.join(getPaths().web.dist, '200.html')

  const fallbackIndexPath = fs.existsSync(prerenderIndexPath)
    ? '200.html'
    : 'index.html'

  // For SPA routing, fallback on unmatched routes and let client-side routing take over
  fastify.setNotFoundHandler({}, (req, reply) => {
    const urlData = req.urlData()
    const requestedExtension = path.extname(urlData.path ?? '')

    // Paths with no extension (`/about`) or an .html extension (`/about.html`)
    // should be handled by the client side router.
    // See the discussion in https://github.com/redwoodjs/redwood/pull/9272.
    if (requestedExtension === '' || requestedExtension === '.html') {
      reply.header('Content-Type', 'text/html; charset=UTF-8')
      return reply.sendFile(fallbackIndexPath)
    }

    // If we got here, the user is requesting an asset with an extension
    // (like `profile.png`) that doesn't exist
    reply.code(404)
    return reply.send('Not Found')
  })

  done()
}

export function resolveOptions(options: RedwoodFastifyWebOptions) {
  const redwood = options.redwood ?? {}

  redwood.apiUrl ??= getConfig().web.apiUrl
  const apiUrlIsFullyQualifiedUrl = isFullyQualifiedUrl(redwood.apiUrl)

  // `apiHost` is deprecated. If it's set and `apiProxyTarget` isn't, we'll use it as `apiProxyTarget`.
  if (redwood.apiHost && !redwood.apiProxyTarget) {
    redwood.apiProxyTarget = redwood.apiHost
    delete redwood.apiHost
  }

  if (redwood.apiProxyTarget && !isFullyQualifiedUrl(redwood.apiProxyTarget)) {
    throw new Error(
      `If you provide \`apiProxyTarget\`, it has to be a fully-qualified URL. \`apiProxyTarget\` is '${redwood.apiProxyTarget}'`
    )
  }

  // If users don't supply `apiUrl` but do supply `apiProxyTarget`, error.
  // We don't have a prefix to use as the starting point of a proxy.
  //
  // ```js
  // {
  //   apiUrl: undefined,
  //   apiProxyTarget: 'http://api.bar.com'
  // }
  // ```
  //
  // This is pretty unlikely because we default `apiUrl` to '/.redwood/functions'
  if (!redwood.apiUrl && redwood.apiProxyTarget) {
    throw new Error(
      `If you provide \`apiProxyTarget\`, \`apiUrl\` has to be a relative URL. \`apiUrl\` is '${redwood.apiUrl}'`
    )
  }

  // If users supply a fully-qualified `apiUrl` and `apiProxyTarget`, error.
  // We don't have a prefix to use as the starting point of a proxy.
  //
  // ```js
  // {
  //   apiUrl: 'http://api.foo.com', // This isn't a prefix we can forward requests from
  //   apiProxyTarget: 'http://api.bar.com'
  // }
  // ```
  if (apiUrlIsFullyQualifiedUrl && redwood.apiProxyTarget) {
    throw new Error(
      `If you provide \`apiProxyTarget\`, \`apiUrl\` cannot be a fully-qualified URL. \`apiUrl\` is '${redwood.apiUrl}'`
    )
  }

  // If users supply a relative `apiUrl` but don't supply `apiProxyTarget`, error.
  // There's nowhere to proxy to.
  //
  // ```js
  // {
  //   apiUrl: '/api',
  //   apiProxyTarget: undefined // There's nowhere for requests to '/api' to go
  // }
  // ```
  // if (!apiUrlIsFullyQualifiedUrl && !redwood.apiProxyTarget) {
  //   throw new Error(
  //     `If you don't provide \`apiProxyTarget\`, \`apiUrl\` needs to be a fully-qualified URL. \`apiUrl\` is '${redwood.apiUrl}'`
  //   )
  // }

  return { redwood }
}

function isFullyQualifiedUrl(url: string) {
  try {
    // eslint-disable-next-line no-new
    new URL(url)
    return true
  } catch (e) {
    return false
  }
}
