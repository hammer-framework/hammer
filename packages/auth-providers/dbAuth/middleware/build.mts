import {
  build,
  defaultBuildOptions,
} from '@redwoodjs/framework-tools'

import { generateCjsTypes} from '@redwoodjs/framework-tools/cjsTypes'
import { writeFileSync } from 'node:fs'

// CJS build
await build({
  buildOptions: {
    ...defaultBuildOptions,
    outdir: 'dist/cjs',
    packages: 'external',
  },
})

// ESM build
await build({
  buildOptions: {
    ...defaultBuildOptions,
    format: 'esm',
    packages: 'external',
  },
})

// Place a package.json file with `type: commonjs` in the dist/cjs folder so that
// all .js files are treated as CommonJS files.
writeFileSync('dist/cjs/package.json', JSON.stringify({ type: 'commonjs' }))

// Place a package.json file with `type: module` in the dist folder so that
// all .js files are treated as ES Module files.
writeFileSync('dist/package.json', JSON.stringify({ type: 'module' }))

await generateCjsTypes()
