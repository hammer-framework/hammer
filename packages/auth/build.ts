import { renameSync, writeFileSync } from 'node:fs'

import { build, defaultBuildOptions } from '@redwoodjs/framework-tools'

// ESM build
await build({
  buildOptions: {
    ...defaultBuildOptions,
    tsconfig: 'tsconfig.build-esm.json',
    format: 'esm',
    outdir: 'dist/esm',
    packages: 'external',
  },
})

// CJS build
await build({
  buildOptions: {
    ...defaultBuildOptions,
    tsconfig: 'tsconfig.build.json',
    packages: 'external',
  },
})

// Because the package.json files has `type: module` the CJS entry file can't
// be named `index.js` because in that case it would be treated as an ESM file.
// By changing it to .cjs it will be treated as a CommonJS file.
renameSync('dist/index.js', 'dist/index.cjs')

// Place a package.json file with `type: commonjs` in the dist folder so that
// all .js files are treated as CommonJS files.
writeFileSync('dist/package.json', JSON.stringify({ type: 'commonjs' }))

// Place a package.json file with `type: module` in the dist/esm folder so that
// all .js files are treated as ES Module files.
writeFileSync('dist/esm/package.json', JSON.stringify({ type: 'module' }))
