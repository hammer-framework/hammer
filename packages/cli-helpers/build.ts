import { build, defaultBuildOptions } from '@redwoodjs/framework-tools'

// ESM build
await build({
  buildOptions: {
    ...defaultBuildOptions,
    bundle: true,
    entryPoints: ['./src/index.ts'],
    format: 'esm',
    outExtension: { '.js': '.mjs' },
    packages: 'external',
  },
})

// CJS build
await build({
  buildOptions: {
    ...defaultBuildOptions,
    bundle: true,
    entryPoints: ['./src/index.ts'],
    outExtension: { '.js': '.cjs' },
    packages: 'external',
  },
})
