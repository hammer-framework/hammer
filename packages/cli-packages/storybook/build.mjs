import fs from 'node:fs'

import * as esbuild from 'esbuild'
import fg from 'fast-glob'

// Get source files
const sourceBinFiles = fg.sync(['./src/bins/**/*.ts'])
const sourceFiles = fg
  .sync(['./src/**/*.ts'])
  .filter((f) => !sourceBinFiles.includes(f))

// Build bin files, we need to insert the shebang in this case
await esbuild.build({
  entryPoints: sourceBinFiles,
  format: 'cjs',
  platform: 'node',
  target: ['node18'],
  outdir: 'dist/bins',
  logLevel: 'info',
  banner: {
    js: '#!/usr/bin/env node',
  },
})

// Build general source files
const result = await esbuild.build({
  entryPoints: sourceFiles,
  format: 'cjs',
  platform: 'node',
  target: ['node18'],
  outdir: 'dist',
  logLevel: 'info',

  // For visualizing the bundle.
  // See https://esbuild.github.io/api/#metafile and https://esbuild.github.io/analyze/.
  metafile: true,
})

fs.writeFileSync('meta.json', JSON.stringify(result.metafile))
