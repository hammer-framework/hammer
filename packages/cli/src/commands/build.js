import fs from 'fs'
import path from 'path'

import execa from 'execa'
import Listr from 'listr'
import VerboseRenderer from 'listr-verbose-renderer'

import { getPaths } from 'src/lib'
import c from 'src/lib/colors'
import { handler as generatePrismaClient } from 'src/commands/dbCommands/generate'

const apiExists = fs.existsSync(getPaths().api.src)
const webExists = fs.existsSync(getPaths().web.src)

const optionDefault = (webExists, apiExists) => {
  let options = []
  if (webExists) options.push('web')
  if (apiExists) options.push('api')
  return options
}

export const command = 'build [app..]'
export const desc = 'Build for production.'
export const builder = {
  app: {
    choices: ['api', 'web'],
    default: optionDefault(webExists, apiExists),
  },
  verbose: { type: 'boolean', default: false, alias: ['v'] },
  stats: { type: 'boolean', default: false },
}

export const handler = async ({
  app = ['api', 'web'],
  verbose = false,
  stats = false,
}) => {
  if (app.includes('api')) {
    try {
      await generatePrismaClient({ verbose, force: true })
    } catch (e) {
      console.log(c.error(e.message))
      process.exit(1)
    }
  }

  const execCommandsForApps = {
    api: {
      // must use path.join() here, and for 'web' below, to support Windows
      cwd: path.join(getPaths().base, 'api'),
      cmd: 'yarn cross-env NODE_ENV=production babel src --out-dir dist',
    },
    web: {
      cwd: path.join(getPaths().base, 'web'),
      cmd: `yarn webpack --config ../node_modules/@redwoodjs/core/config/webpack.${
        stats ? 'stats' : 'production'
      }.js`,
    },
  }

  if (stats) {
    app = ['web']
    console.log(
      ' ⏩ Skipping `api` build because stats only works for Webpack at the moment.'
    )
  }

  const tasks = new Listr(
    app.map((appName) => {
      const { cwd, cmd } = execCommandsForApps[appName]
      return {
        title: `Building "${appName}"${(stats && ' for stats') || ''}...`,
        task: () => {
          return execa(cmd, undefined, {
            stdio: verbose ? 'inherit' : 'pipe',
            shell: true,
            cwd,
          })
        },
      }
    }),
    {
      renderer: verbose && VerboseRenderer,
    }
  )

  try {
    await tasks.run()
  } catch (e) {
    console.log(c.error(e.message))
    process.exit(1)
  }
}
