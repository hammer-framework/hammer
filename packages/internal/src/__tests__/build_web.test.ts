import path from 'path'

import { prebuildWebFiles, cleanWebBuild } from '../build/web'
import { findWebFiles } from '../files'
import { ensurePosixPath } from '../paths'

const FIXTURE_PATH = path.resolve(
  __dirname,
  '../../../../__fixtures__/example-todo-main'
)

const cleanPaths = (p) => {
  return ensurePosixPath(path.relative(FIXTURE_PATH, p))
}

const fullPath = (p) => {
  return path.join(FIXTURE_PATH, p)
}

// Fixtures, filled in beforeAll
let prebuiltFiles
let relativePaths

beforeAll(() => {
  process.env.RWJS_CWD = FIXTURE_PATH
  cleanWebBuild()

  const webFiles = findWebFiles()
  prebuiltFiles = prebuildWebFiles(webFiles)

  relativePaths = prebuiltFiles
    .filter((x) => typeof x !== 'undefined')
    .map(cleanPaths)
})
afterAll(() => {
  delete process.env.RWJS_CWD
})

test('api files are prebuilt', () => {
  // Builds non-nested functions
  relativePaths //?
})
