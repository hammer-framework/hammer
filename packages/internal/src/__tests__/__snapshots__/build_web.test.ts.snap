// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check routes are imported with "import" when staticImports flag is NOT passed 1`] = `
"\\"use strict\\";

require(\\"core-js/modules/es.object.define-property.js\\");

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports[\\"default\\"] = void 0;

require(\\"core-js/modules/es.array.iterator.js\\");

require(\\"core-js/modules/es.object.to-string.js\\");

require(\\"core-js/modules/es.set.js\\");

require(\\"core-js/modules/es.string.iterator.js\\");

require(\\"core-js/modules/esnext.set.add-all.js\\");

require(\\"core-js/modules/esnext.set.delete-all.js\\");

require(\\"core-js/modules/esnext.set.difference.js\\");

require(\\"core-js/modules/esnext.set.every.js\\");

require(\\"core-js/modules/esnext.set.filter.js\\");

require(\\"core-js/modules/esnext.set.find.js\\");

require(\\"core-js/modules/esnext.set.intersection.js\\");

require(\\"core-js/modules/esnext.set.is-disjoint-from.js\\");

require(\\"core-js/modules/esnext.set.is-subset-of.js\\");

require(\\"core-js/modules/esnext.set.is-superset-of.js\\");

require(\\"core-js/modules/esnext.set.join.js\\");

require(\\"core-js/modules/esnext.set.map.js\\");

require(\\"core-js/modules/esnext.set.reduce.js\\");

require(\\"core-js/modules/esnext.set.some.js\\");

require(\\"core-js/modules/esnext.set.symmetric-difference.js\\");

require(\\"core-js/modules/esnext.set.union.js\\");

require(\\"core-js/modules/web.dom-collections.iterator.js\\");

var _react = _interopRequireDefault(require(\\"react\\"));

var _router = require(\\"@redwoodjs/router\\");

var _SetLayout = _interopRequireDefault(require(\\"./layouts/SetLayout/SetLayout\\"));

var _FooPage = _interopRequireDefault(require(\\"./pages/FooPage/FooPage\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { \\"default\\": obj }; }

const BarPage = {
  name: \\"BarPage\\",
  loader: () => import(\\"./pages/BarPage/BarPage\\")
};
const FatalErrorPage = {
  name: \\"FatalErrorPage\\",
  loader: () => import(\\"./pages/FatalErrorPage/FatalErrorPage\\")
};
const HomePage = {
  name: \\"HomePage\\",
  loader: () => import(\\"./pages/HomePage/HomePage\\")
};
const NotFoundPage = {
  name: \\"NotFoundPage\\",
  loader: () => import(\\"./pages/NotFoundPage/NotFoundPage\\")
};
const PrivatePage = {
  name: \\"PrivatePage\\",
  loader: () => import(\\"./pages/PrivatePage/PrivatePage\\")
};
const TypeScriptPage = {
  name: \\"TypeScriptPage\\",
  loader: () => import(\\"./pages/TypeScriptPage/TypeScriptPage\\")
};
const adminEditUserPage = {
  name: \\"adminEditUserPage\\",
  loader: () => import(\\"./pages/admin/EditUserPage/EditUserPage\\")
};

var Routes = function Routes() {
  return /*#__PURE__*/_react[\\"default\\"].createElement(_router.Router, null, /*#__PURE__*/_react[\\"default\\"].createElement(_router.Route, {
    path: \\"/\\",
    page: HomePage,
    name: \\"home\\",
    prerender: true
  }), /*#__PURE__*/_react[\\"default\\"].createElement(_router.Route, {
    path: \\"/typescript\\",
    page: TypeScriptPage,
    name: \\"typescriptPage\\",
    prerender: true
  }), /*#__PURE__*/_react[\\"default\\"].createElement(_router.Route, {
    path: \\"/somewhereElse\\",
    page: EditUserPage,
    name: \\"someOtherPage\\",
    prerender: true
  }), /*#__PURE__*/_react[\\"default\\"].createElement(Set, {
    wrap: _SetLayout[\\"default\\"],
    prerender: true
  }, /*#__PURE__*/_react[\\"default\\"].createElement(_router.Route, {
    path: \\"/foo\\",
    page: _FooPage[\\"default\\"],
    name: \\"fooPage\\"
  }), /*#__PURE__*/_react[\\"default\\"].createElement(_router.Route, {
    path: \\"/bar\\",
    page: BarPage,
    name: \\"barPage\\"
  })), /*#__PURE__*/_react[\\"default\\"].createElement(_router.Private, {
    prerender: true,
    unauthenticated: \\"home\\"
  }, /*#__PURE__*/_react[\\"default\\"].createElement(_router.Route, {
    path: \\"/private-page\\",
    page: PrivatePage,
    name: \\"privatePage\\"
  })), /*#__PURE__*/_react[\\"default\\"].createElement(_router.Route, {
    notfound: true,
    page: NotFoundPage
  }));
};

var _default = Routes;
exports[\\"default\\"] = _default;"
`;

exports[`Check routes are imported with require when staticImports flag is enabled 1`] = `
"\\"use strict\\";

require(\\"core-js/modules/es.object.define-property.js\\");

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports[\\"default\\"] = void 0;

require(\\"core-js/modules/es.array.iterator.js\\");

require(\\"core-js/modules/es.object.to-string.js\\");

require(\\"core-js/modules/es.set.js\\");

require(\\"core-js/modules/es.string.iterator.js\\");

require(\\"core-js/modules/esnext.set.add-all.js\\");

require(\\"core-js/modules/esnext.set.delete-all.js\\");

require(\\"core-js/modules/esnext.set.difference.js\\");

require(\\"core-js/modules/esnext.set.every.js\\");

require(\\"core-js/modules/esnext.set.filter.js\\");

require(\\"core-js/modules/esnext.set.find.js\\");

require(\\"core-js/modules/esnext.set.intersection.js\\");

require(\\"core-js/modules/esnext.set.is-disjoint-from.js\\");

require(\\"core-js/modules/esnext.set.is-subset-of.js\\");

require(\\"core-js/modules/esnext.set.is-superset-of.js\\");

require(\\"core-js/modules/esnext.set.join.js\\");

require(\\"core-js/modules/esnext.set.map.js\\");

require(\\"core-js/modules/esnext.set.reduce.js\\");

require(\\"core-js/modules/esnext.set.some.js\\");

require(\\"core-js/modules/esnext.set.symmetric-difference.js\\");

require(\\"core-js/modules/esnext.set.union.js\\");

require(\\"core-js/modules/web.dom-collections.iterator.js\\");

var _react = _interopRequireDefault(require(\\"react\\"));

var _router = require(\\"@redwoodjs/router\\");

var _SetLayout = _interopRequireDefault(require(\\"./layouts/SetLayout/SetLayout\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { \\"default\\": obj }; }

const BarPage = {
  name: \\"BarPage\\",
  loader: () => require(\\"./pages/BarPage/BarPage\\")
};
const FatalErrorPage = {
  name: \\"FatalErrorPage\\",
  loader: () => require(\\"./pages/FatalErrorPage/FatalErrorPage\\")
};
const FooPage = {
  name: \\"FooPage\\",
  loader: () => require(\\"./pages/FooPage/FooPage\\")
};
const HomePage = {
  name: \\"HomePage\\",
  loader: () => require(\\"./pages/HomePage/HomePage\\")
};
const NotFoundPage = {
  name: \\"NotFoundPage\\",
  loader: () => require(\\"./pages/NotFoundPage/NotFoundPage\\")
};
const PrivatePage = {
  name: \\"PrivatePage\\",
  loader: () => require(\\"./pages/PrivatePage/PrivatePage\\")
};
const TypeScriptPage = {
  name: \\"TypeScriptPage\\",
  loader: () => require(\\"./pages/TypeScriptPage/TypeScriptPage\\")
};
const adminEditUserPage = {
  name: \\"adminEditUserPage\\",
  loader: () => require(\\"./pages/admin/EditUserPage/EditUserPage\\")
};

var Routes = function Routes() {
  return /*#__PURE__*/_react[\\"default\\"].createElement(_router.Router, null, /*#__PURE__*/_react[\\"default\\"].createElement(_router.Route, {
    path: \\"/\\",
    page: HomePage,
    name: \\"home\\",
    prerender: true
  }), /*#__PURE__*/_react[\\"default\\"].createElement(_router.Route, {
    path: \\"/typescript\\",
    page: TypeScriptPage,
    name: \\"typescriptPage\\",
    prerender: true
  }), /*#__PURE__*/_react[\\"default\\"].createElement(_router.Route, {
    path: \\"/somewhereElse\\",
    page: EditUserPage,
    name: \\"someOtherPage\\",
    prerender: true
  }), /*#__PURE__*/_react[\\"default\\"].createElement(Set, {
    wrap: _SetLayout[\\"default\\"],
    prerender: true
  }, /*#__PURE__*/_react[\\"default\\"].createElement(_router.Route, {
    path: \\"/foo\\",
    page: FooPage,
    name: \\"fooPage\\"
  }), /*#__PURE__*/_react[\\"default\\"].createElement(_router.Route, {
    path: \\"/bar\\",
    page: BarPage,
    name: \\"barPage\\"
  })), /*#__PURE__*/_react[\\"default\\"].createElement(_router.Private, {
    prerender: true,
    unauthenticated: \\"home\\"
  }, /*#__PURE__*/_react[\\"default\\"].createElement(_router.Route, {
    path: \\"/private-page\\",
    page: PrivatePage,
    name: \\"privatePage\\"
  })), /*#__PURE__*/_react[\\"default\\"].createElement(_router.Route, {
    notfound: true,
    page: NotFoundPage
  }));
};

var _default = Routes;
exports[\\"default\\"] = _default;"
`;
